src/components/Media.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { maybeSignUri } from '@/handlers/imgix';
+import { Image, ImageStyle, View } from 'react-native';
+// @ts-ignore
+import Video from 'react-native-video';
+import SvgImage from './svg/SvgImage';
+
+export enum MimeType {
+  MP4 = 'video/mp4',
+  GIF = 'image/gif',
+  PNG = 'image/png',
+  SVG = 'image/svg+xml',
+  JPG = 'image/jpeg',
+}
+
+export function Media({
+  url,
+  mimeType,
+  fallbackUrl,
+  style,
+  size,
+  onLayout,
+  onError,
+}: {
+  url: string;
+  mimeType?: string;
```

Comment:
should we type this as MimeType? could see us using for diff providers that may not overlap fully with the types

Reply:
while mime type values _should_ be universal, i don't think we should type mimeType as MimeType here bc apis like reservoir/simplehash/mint.fun are just going to type their mimeType attributes as strings. so unless we actually go out of our way to make safe conversions from string -> MimeType, we'll just be forced to do type assertions when passing in mimeType to Media as a prop